src/core/exec-invoke.c

```diff
@@ -5441,6 +5525,14 @@ int exec_invoke(
         if (r < 0)
                 return r;
 
+        if (context->private_bpf) {
+                r = bpffs_finalize(bpffs_fd, TAKE_PIDREF(bpffs_pid).pid);
```

Comment:
If `bpffs_finalize()` fails earlier, the child process may not be sigkilled.

Reply:
because of TAKE_PIDREF()?

Reply:
Right.